packages/server/src/mL/index.ts

```diff
@@ -35,14 +36,24 @@ export default class MLService {
 
     await this.docker.reset();
 
-    this.dockerState = DockerState.TrainPull;
-    await this.docker.pullImages(Object.values(Trainer.images));
-
-    this.dockerState = DockerState.ExportPull;
-    await this.docker.pullImages(Object.values(Exporter.images));
-
-    this.dockerState = DockerState.TestPull;
-    await this.docker.pullImages(Object.values(Tester.images));
+    https
+      .get("https://hub.docker.com/", async (res) => {
+        if (res.statusCode === 200) {
+          this.dockerState = DockerState.TrainPull;
+          await this.docker.pullImages(Object.values(Trainer.images));
+
+          this.dockerState = DockerState.ExportPull;
+          await this.docker.pullImages(Object.values(Exporter.images));
+
+          this.dockerState = DockerState.TestPull;
+          await this.docker.pullImages(Object.values(Tester.images));
+        } else {
+          console.log("No internet, skipping pull containers");
```

Comment:
What happens if the user does not have the containers already?

Reply:
+1

Reply:
You can no longer get to this point without all the images.